### PR TITLE
Try to make create/edit process a bit more robust

### PR DESCRIPTION
**Pull request by travjenkins**
## Issues

https://github.com/estuary/ui/issues/913

### 913
- Remove the Test and Save buttons after a successful publish and replace with a 'View Details` button that takes the user to the details
- Added some error handling for super edge cases where the publish might have a call failed during it.
- Updating the `Test` and `Save & Publish` buttons to show `Testing` and `Saving & Publishing` while there is an action.
- Added `Saved` to the active state so forms cannot be changed after being saved.

### misc
- Update Icon Noir. This was initially part of an experiment but the work is valid so leaving it in
- Adding in a disabled state for remove for when it is removing
- Added a check that the call to the server to remove the binding actually worked before removing it from the store
- Made the removal of bindings a bit more robust to potential errors. This really only happened if a user would click on the remove button of multiple bindings quickly. 
- Fixed some random DOM issues that Chrome called out where `id`s and `name`s were missing on inputs/buttons.
- Fixed issue where time travel was editable during an async action to the server

### Manually tested

-   scenarios you manually tested

### Automated tests

-   unit testing covered

## Screenshots

_If applicable - please include some screenshots of the new UI_

### CODE REVIEW

File: src/components/capture/useCaptureDraftUpdate.ts [hunk 1]
```diff
@@ -115,6 +115,17 @@ function useDiscoverDraftUpdate(options?: {
...
 
+                if (draftSpecsResponse.data.length === 0) {
+                    callFailed({
+                        error: {
+                            title: 'captureCreate.generate.failedErrorTitle',
+                        },
+                    });
+
+                    return false;
+                }
```
Comment (travjenkins) [inline src/components/capture/useCaptureDraftUpdate.ts:126]: Rare - but wanted to handle this error just in case.

File: src/utils/publication-utils.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { logRocketConsole } from 'services/shared';
+import { JOB_STATUS_SUCCESS } from 'services/supabase';
+import { hasLength } from './misc-utils';
+
+export const checkIfPublishIsDone = (payload: any): [boolean | null, any] => {
```
Comment (travjenkins) [inline src/utils/publication-utils.ts:5]: This was broken out for stuff that was reverted - but leaving it stand alone makes sense.

File: src/components/collection/Selector/List/index.tsx [hunk 3]
```diff
@@ -102,9 +102,8 @@ function CollectionSelectorList({
     const formStatus = useFormStateStore_status();
 
     const selectionEnabled =
-        currentBindingUUID &&
-        setCurrentBinding &&
-        formStatus !== FormStatus.UPDATING;
+        // currentBindingUUID &&
```
Comment (kiahna-tucker) [inline src/components/collection/Selector/List/index.tsx:105]: This commented out line of code can be removed outright if it is no longer needed.

File: src/components/shared/Entity/Actions/SaveButton.tsx [hunk 4]
```diff
@@ -36,6 +36,8 @@ function EntitySaveButton({ disabled, taskNames, logEvent }: Props) {
     const logToken = useFormStateStore_logToken();
     const formStatus = useFormStateStore_status();
 
+    const isSaving = formStatus === FormStatus.SAVING;
```
Comment (kiahna-tucker) [inline src/components/shared/Entity/Actions/SaveButton.tsx:39]: This variable naming convention is _okay_ but, from my vantage point, _is_ is a redundant prefix when the variable name is a verb with an _-ing_ suffix.

File: src/components/shared/Entity/HeaderLogs.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    useFormStateStore_messagePrefix,
+    useFormStateStore_showLogs,
+    useFormStateStore_status,
+} from 'stores/FormState/hooks';
+import { FormStatus } from 'stores/FormState/types';
+import useEntityWorkflowHelpers from './hooks/useEntityWorkflowHelpers';
+import LogDialog from './LogDialog';
+import LogDialogActions from './LogDialogActions';
+
+// TODO (break logs from buttons) - this is starting to display the dialog as a stand alone
```
Comment (kiahna-tucker) [inline src/components/shared/Entity/HeaderLogs.tsx:13]: Assuming this TODO comment identifies a task out of scope for this PR.

File: src/stores/FormState/Store.ts [hunk 6]
```diff
@@ -14,7 +14,10 @@ const formActive = (status: FormStatus) => {
         status === FormStatus.GENERATING ||
         status === FormStatus.SAVING ||
         status === FormStatus.SCHEMA_EVOLVING ||
-        status === FormStatus.UPDATING
+        status === FormStatus.UPDATING ||
+        // TODO (workflow stores) need to manage form state better
+        // This is crappy - worry. But if we have saved we want to disable everything and this is quickest way
```
Comment (kiahna-tucker) [inline src/stores/FormState/Store.ts:19]: > This is crappy - worry.

I imagine _worry_ is a typo and you meant to say _sorry._

File: src/stores/Binding/Store.ts [hunk 7]
```diff
@@ -1043,13 +1049,27 @@ const getInitialState = (
                     ? state.resourceConfigs[targetBindingUUID]
                     : null;
 
+                const targettedResourceConfig =
```
Comment (kiahna-tucker) [inline src/stores/Binding/Store.ts:1052]: This variable could simply be `targetResourceConfig` since it is the compliment of `targetBindingUUID`.

**Review by kiahna-tucker — COMMENTED** [review]
Posting the code review comments while I test the PR.

File: src/components/shared/Entity/hooks/useEntityWorkflowHelpers.ts [hunk 8]
```diff
@@ -142,21 +143,30 @@ function useEntityWorkflowHelpers() {
...
         },
         [catalogName, generatePath, navigate, resetState]
     );
 
     // Form Event Handlers
-    const closeLogs = useCallback(() => {
-        setFormState({
-            showLogs: false,
-        });
+    const closeLogs = useCallback(
```
Comment (kiahna-tucker) [inline src/components/shared/Entity/hooks/useEntityWorkflowHelpers.ts:153]: This function will need to be explicitly called wherever it is used (i.e., `LogDialogActions`, `HeaderLogs` [unused component]) since an input argument has been introduced. Otherwise, `navigateOnly` will be a button event object and, thus, always truthy.

File: src/components/shared/Entity/HeaderLogs.tsx (hunk 5 above)
Comment (travjenkins) [reply src/components/shared/Entity/HeaderLogs.tsx:13]: Yeah... I didn't want to lose this component as it is really nice to help with extra renders but also didn't want an old file floating around.

**Review by kiahna-tucker — CHANGES_REQUESTED** [review]
The main issue is identified in a comment below.

And did we ultimately decide to no longer allow the _Close_ CTA in the publication dialog to navigate the user to the _Details_ page when clicked after the publication job completed? The _View Details_ CTA was more an alternate means to navigate to the _Details_ page in my recollection, to be used when a user closes the publication dialog before the publication job completes.

File: src/components/collection/Selector/List/index.tsx (hunk 3 above)
Comment (travjenkins) [reply src/components/collection/Selector/List/index.tsx:105]: Good catch... this needs added back in. This was a part of testing out not setting a default binding to show.

File: src/components/shared/Entity/hooks/useEntityWorkflowHelpers.ts (hunk 8 above)
Comment (travjenkins) [reply src/components/shared/Entity/hooks/useEntityWorkflowHelpers.ts:153]: dang it... yeah... I'll just take this out as it was a quick add at the end to try to make things better

Comment (travjenkins) [reply src/components/shared/Entity/hooks/useEntityWorkflowHelpers.ts:153]: also... this was _really_ dumb since we already return `exit`

File: src/components/shared/Entity/Actions/SaveButton.tsx (hunk 4 above)
Comment (travjenkins) [reply src/components/shared/Entity/Actions/SaveButton.tsx:39]: I'm fine either way but this followed the `isTesting` in the other file. I could go either way with removing the `is` from booleans denoting a verb-ing.

Comment (travjenkins) [reply src/components/shared/Entity/Actions/SaveButton.tsx]: I forgot to use this variable in the logic right below... switched out the name.

File: src/stores/Binding/Store.ts [hunk 9]
```diff
@@ -658,15 +660,19 @@ const getInitialState = (
...
+                            evaluatedBindingUUID && evaluatedResourceConfig
+                        )
+                            ? {
+                                  uuid: evaluatedBindingUUID,
+                                  collection:
+                                      evaluatedResourceConfig.meta
+                                          .collectionName,
+                              }
+                            : null;
+                    }
```
Comment (travjenkins) [inline src/stores/Binding/Store.ts:675]: We used to keep the current selected during removing... I think that works a bit better. So putting that back in.

File: src/stores/Binding/Store.ts [hunk 10]
```diff
@@ -1043,13 +1049,27 @@ const getInitialState = (
...
+                    logRocketEvent(
+                        CustomEvents.BINDINGS_RESOURCE_CONFIG_MISSING,
+                        {
+                            targetBindingUUID,
+                            targetCollection,
+                        }
+                    );
+
+                    return;
+                }
```
Comment (travjenkins) [inline src/stores/Binding/Store.ts:1066]: This should not really happen and it was really hard to reproduce but putting this here and hoping we can see if this keeps happening.

**Review by kiahna-tucker — APPROVED** [review]
Approved with moderate testing. Primarily retested the erring scenario noted in the previous review comment in all affected workflows.
